[web-animations] mark AnimationEffect subclasses as final

graouts opened the pull request:
#### b84c65e04ccc71499331ce4467bbb409860cc5f6
<pre>
[web-animations] mark AnimationEffect subclasses as final
<a href="https://bugs.webkit.org/show_bug.cgi?id=252051">https://bugs.webkit.org/show_bug.cgi?id=252051</a>

Reviewed by Dean Jackson.

We also mark the few overridden methods previously marked as `override`
as `final` to harmonize with the way other overriden methods are marked
in those files.

* Source/WebCore/animation/CustomEffect.h:
* Source/WebCore/animation/KeyframeEffect.h:

Canonical link: <a href="https://commits.webkit.org/260112@main">https://commits.webkit.org/260112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1201629e1f50d6d4079dfc125e8c8626cf8a9afc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/107117 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/16167 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39909 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/116293 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115781 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111019 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17628 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/7367 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/99297 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112882 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/13319 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/96374 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40928 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/95217 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/28010 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82689 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/9255 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/29352 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9864 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/6347 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/15413 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48928 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/11405 "Built successfully") | | | 
| [❌ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3776 "Failed to push commit to Webkit repository") | | | | 
<!--EWS-Status-Bubble-End-->

maattdd commented on Source/WebCore/animation/KeyframeEffect.h:
```diff
@@ -132,9 +132,9 @@ class KeyframeEffect : public AnimationEffect, public CSSPropertyBlendingClient
 
-    Document* document() const override;
-    RenderElement* renderer() const override;
-    const RenderStyle& currentStyle() const override;
```
Why are we removing the `override` ? 

`override final` would be safer

graouts replied:
I won't argue for or against `override final` but there is no use of `override final` in the entire WebKit project. I don't think we have a policy to use either `override` or `final` in classes marked `final`, I was only making that change for consistency within that file which uses `final` instead of `override` in the vast majority of cases.

graouts committed:
[web-animations] mark AnimationEffect subclasses as final
https://bugs.webkit.org/show_bug.cgi?id=252051

Reviewed by Dean Jackson.

We also mark the few overridden methods previously marked as `override`
as `final` to harmonize with the way other overriden methods are marked
in those files.

* Source/WebCore/animation/CustomEffect.h:
* Source/WebCore/animation/KeyframeEffect.h:

Canonical link: https://commits.webkit.org/260112@main
